### PR TITLE
Add `blockchain.getDifficulty` RPC

### PR DESCRIPTION
**Pull request by izzycsy**
Add getDifficulty

#### What kind of PR is this?:

/kind feature
Add getDifficulty rpc

### CODE REVIEW

File: website/docs/jellyfish/api/blockchain.md [hunk 1]
```diff
@@ -188,6 +188,14 @@ interface ChainTip {
...
 }
 ```
+## getDifficulty
+Return the proof-of-work difficulty as a multiple of the minimum difficulty.
+```ts tile="client.blockchain.getDifficulty()"
+interface difficulty {
+    getDifficulty (): Promise<number>
+}
+```
+
```
Comment (fuxingloh) [inline website/docs/jellyfish/api/blockchain.md:198]: Keep in mind of code line breaks

```suggestion

## getDifficulty

Return the proof-of-work difficulty as a multiple of the minimum difficulty.

```ts tile="client.blockchain.getDifficulty()"
interface blockchain {
    getDifficulty (): Promise<number>
}
```
```

File: packages/jellyfish-api-core/src/category/blockchain.ts [hunk 2]
```diff
@@ -110,6 +110,14 @@ export class Blockchain {
     return await this.client.call('getchaintips', [], 'number')
   }
 
+  /**
+   * Get difficulty of everything.
+   * Return difficulty as number.
+   */
```
Comment (fuxingloh) [inline packages/jellyfish-api-core/src/category/blockchain.ts:116]:
```suggestion
  /**
   * Get difficulty of chain.
   *
   * @return {Promise<number>}
   */
```

Comment (fuxingloh) [reply packages/jellyfish-api-core/src/category/blockchain.ts:116]: JSDocs need to be structured properly

File: packages/jellyfish-api-core/__tests__/category/blockchain.test.ts [hunk 3]
```diff
@@ -249,6 +249,27 @@ describe('masternode', () => {
...
+    beforeAll(async () => {
+      await container.start()
+      await container.waitForReady()
+      // await container.waitForWalletCoinbaseMaturity()
+    })
+
+    afterAll(async () => {
+      await container.stop()
+    })
+
```
Comment (fuxingloh) [inline packages/jellyfish-api-core/__tests__/category/blockchain.test.ts:266]: Remove this entire code section, it's not required. it's done by the parent `describe`.

File: packages/jellyfish-api-core/__tests__/category/blockchain.test.ts [hunk 4]
```diff
@@ -249,6 +249,27 @@ describe('masternode', () => {
...
+    it('should getDifficulty', async () => {
+      const getdifficulty = await client.blockchain.getDifficulty()
+      expect(getdifficulty).toBeGreaterThanOrEqual(0)
```
Comment (fuxingloh) [inline packages/jellyfish-api-core/__tests__/category/blockchain.test.ts:269]:
```suggestion
      const difficulty = await client.blockchain.getDifficulty()
      expect(difficulty).toBeGreaterThanOrEqual(0)
```

File: packages/jellyfish-api-core/__tests__/category/blockchain.test.ts [hunk 5]
```diff
@@ -249,6 +249,27 @@ describe('masternode', () => {
     })
   })
 
+  // getDifficulty of masternode & non masternode
```
Comment (fuxingloh) [inline packages/jellyfish-api-core/__tests__/category/blockchain.test.ts:252]: remove this line

**Review by fuxingloh — CHANGES_REQUESTED** [review]

File: packages/jellyfish-api-core/src/category/blockchain.ts [hunk 6]
```diff
@@ -110,6 +110,15 @@ export class Blockchain {
     return await this.client.call('getchaintips', [], 'number')
   }
 
+  /**
+   * Get difficulty of chain.
```
Comment (fuxingloh) [inline packages/jellyfish-api-core/src/category/blockchain.ts:114]: This should be 
```
Get the proof-of-work difficulty as a multiple of the minimum difficulty.
```

**Review by fuxingloh — APPROVED** [review]

**Review by thedoublejay — APPROVED** [review]

**Pull request merged**